docs/reference/modules/cluster/misc.asciidoc

```diff
@@ -101,6 +101,25 @@ number of shards for each node, use the
 setting.
 --
 
+[[bounded-diagnostic-requests-limit]]
+===== Bounded diagnostic requests limit
+
+A diagnostic request might require information from all nodes to be aggregated before it returns to the user.
+These calls can be heavy and they put extra pressure on the coordinating node (the node collecting the responses from all the nodes), for this reason there is a limit on the concurrent requests that a node can coordinate.
+Currently, the only bounded diagnostic requests are <<indices-stats, index stats>>
+and <<indices-recovery, recovery>> APIs.
+
+--
+
+[[cluster-max-concurrent-bounded-diagnostic-requests-per-node]]
+`cluster.max_concurrent_bounded_diagnostic_requests_per_node`::
++
+--
+(<<dynamic-cluster-setting,Dynamic>>)
+Limits the bounded diagnostic requests a coordinating node can concurrently handle.
```

Comment:
It would be good to include what the current default is here also, I think

Reply:
Definitely, I will add it.